Introduction to Free Monads/article.md

```diff
@@ -84,8 +84,42 @@ Then `mempty = []` and `(<>) = (++)`.
 $M$ contains `[]`, as required by `mempty`, all single-element lists, and all possible concatenations.
 Finally, all the monoid laws hold, and no other laws are implied.
 
-On the other hand, addition over non-negative integers doesn't constitute a free monoid.
-That would imply the law of commutativity, which isn't a monoid law.
+On the other hand, addition over integers doesn't constitute a free monoid.
+That would imply the law of commutativity, which isn't a monoid law, and, in case of integers, doesn't follow from monoid laws.
+
+<p>
+	<details>
+	<summary>On non-free monoids</summary>
+
+The previous version of this article incorrectly claimed that addition over _non-negative_ integers doesn't constitute a free monoid (thanks Michał Kukieła for the correction!).
+The reason for why that is incorrect is that for naturals, commutativity immediately follows from associativity:
+
+$\begin{align}
+& n+k
+\\ & = \underset{n \text{ times}}{\underbrace{(1 + \ldots + 1)}} + \underset{k \text{ times}}{\underbrace{(1 + \ldots + 1)}}
+\\ & = \underset{k \text{ times}}{\underbrace{(1 + \ldots + 1)}} + \underset{n \text{ times}}{\underbrace{(1 + \ldots + 1)}}
+\\ & = k + n
+\end{align}$
+
+This doesn't hold if integers can be of different sign.
+To see why, assume, e.g. $n < 0 < k:$ in this case, it's impossible to get $k+n$ from $n+k$ just by reshuffling parentheses.
+
+Another way to look at it is this: a free monoid doesn't do anything interesting.
+Hence, we can "recover" any other monoid from a free one.
+
+More formally, if $M$ is a free monoid over $S$, then for any monoid $N$, given a map $f : S \to N$, we can extend this mapping to a monoid morphism $f' : M \to N.$
```

Comment:
```suggestion
More formally, if $M$ is a free monoid over $S$, then for any monoid $N$, given a map $f : S \to N$, we can extend this mapping to a monoid morphism $f' : M \to N$ in a unique way.
```

The uniqueness is needed for the implication in line 112. I think it does not hurt to add it.